CompetitionTemplate/src/competition/subsystems/vision/JetsonCommPacket.java

```diff
@@ -0,0 +1,190 @@
+package competition.subsystems.vision;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.stream.Collectors;
+
+import org.apache.log4j.Logger;
+
+/**
+ * Parses and stores data from incoming vision packets.
+ *
+ */
+public class JetsonCommPacket {
+    static Logger log = Logger.getLogger(JetsonCommPacket.class);
+    
+    private static final int startOfPacketInt = Integer.MAX_VALUE;
+
+    private PacketParserState currentParserState = PacketParserState.WAITING_FOR_START;
+
+    private ArrayList<Integer> packetPayloadData = new ArrayList<>();
+    private int expectedPayloadLength = -1;
+    private PacketPayloadType payloadType = PacketPayloadType.UNKNOWN;
+
+    private double reciptTimestamp = Double.NEGATIVE_INFINITY;
+    
+    public JetsonCommPacket() {
+
+    }
+    
+    public JetsonCommPacket(int[] data) {
+        if(addNewValues(data)) {
+            log.warn("Packet object initialized with partial packet data!");
+        }
+    }
+
+    /*
+     * Packets are sets of 32-bit (4-byte) signed integers, formatted as follows.
+     * 
+     * [0] Packet start flag. Equivalent to Integer.MAX_VALUE.
+     * [1] Payload type flag. Integer specifying the type of data, as described in PacketPayloadType.
+     * [2] Packet length. Length of packet, indicating the number of 32-bit integers in the payload.
+     * [3..packet length + 2] Payload data. Type-dependent.
+     */
+
+    /**
+     * Processes a newly-received integer value and updates internal state tracking accordingly.
+     * @param newValue the new value to process.
+     * @return A boolean indicating whether more data are needed to fill this packet.
+     */
+    public boolean addNewValue(int newValue) {
+        switch (currentParserState) {
+            case WAITING_FOR_START:
+                // Skip data until we find the start of a packet
+                if (newValue == startOfPacketInt) {
+                    this.currentParserState = PacketParserState.WAITING_FOR_PAYLOAD_TYPE_FLAG;
+                }
+                break;
+
+            case WAITING_FOR_PAYLOAD_TYPE_FLAG:
+                this.payloadType = PacketPayloadType.parse(newValue);
+                
+                if(this.payloadType == null) {
+                    currentParserState = PacketParserState.MALFORMED_PACKET_ABORT;
+                    // If we get junky data, give up
```

Comment:
Maybe, but not here. If we want to separately handle malformed packets, I think we should do it in the `JetsonServerManager`.